apps/autoscheduler/ThroughputPredictor.h

```diff
@@ -0,0 +1,25 @@
+#include "HalideBuffer.h"
+
+namespace Halide {
+class ThroughputPredictor {
+ public:
+  virtual ~ThroughputPredictor() { }
+  virtual void set_pipeline_features(const Runtime::Buffer<float> &pipeline_feats, int n) = 0;
+  virtual void enqueue(int ns, Runtime::Buffer<float> *schedule_feats, double *cost_ptr) = 0;
+  virtual void evaluate_costs() = 0;
+  virtual void reset() = 0;
+};
+
+
+class ThroughputPredictorPipeline : public ThroughputPredictor {
```

Comment:
I'm confused about this derived class. It appears to do nothing but has the same name as ThroughputPredictorPipeline below

Reply:
I'm not using it yet, but I will in my follow up set of changes. I'll move some of the code that search through the set of possible schedule into a separate file. That code will take a ThroughputPredictor instead of a ThroughputPredictorPipeline in argument. That will allow me to plug other neural network architectures in the scheduler. The code in AutoSchedule.cpp will continue to use the ThroughputPredictorPipeline since that's the best we have atm.